specification/security/resource-manager/Microsoft.Security/preview/2020-08-06-preview/insights.json

```diff
@@ -0,0 +1,187 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Security Center",
+    "description": "API spec for Microsoft.Security (Azure Security Center) resource provider",
+    "version": "2020-08-06-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{resourceId}/providers/Microsoft.Security/insights": {
+      "get": {
+        "x-ms-examples": {
+          "List insights of an Azure storage resource": {
+            "$ref": "./examples/Insights/ListStorageInsights_example.json"
+          }
+        },
+        "tags": [
+          "Insights"
+        ],
+        "description": "Insights list of a specific resource",
+        "operationId": "Insights_List",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/InsightsList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{resourceId}/providers/Microsoft.Security/insights/{insightName}": {
+      "get": {
+        "x-ms-examples": {
+          "Get classification insights of an Azure storage resource": {
+            "$ref": "./examples/Insights/GetStorageClassificationInsights_example.json"
+          }
+        },
+        "tags": [
+          "Insights"
+        ],
+        "description": "Insights of a specific resource",
+        "operationId": "Insights_Get",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/InsightName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/InsightsModel"
```

Comment:
Please keep in mind that ARM has a size limit on the response payload. Based on the current modeling, if the response size continues to grow, it will problematic. I see there are things like catalogs and labels. not sure how these are created, but can this result in a huge payload for a certain scope? If yes, may be you can create child resource types for those under insights. The insights can return some common properties and rest is via the child type which can be paginated etc since it will become a resource. Does this usecase make sense? please think about it if it can be valid.
Do the customers get insight over a resource like VM or DB or can they get it for the subscription?

Reply:
@ravbhatnagar The response is aggregative and the number of classifications/labels is relatively small.
We don't expect to reach large payloads that will exceed the size limit that ARM has (Not even close to the limit).
We'll keep monitoring this and make sure that this does not happen.